ush/ufsda/yamltools.py

```diff
@@ -128,19 +168,32 @@ def calc_time_vars(config):
     bkg_time_obj = valid_time_obj
     config['BKG_YYYYmmddHHMMSS'] = bkg_time_obj.strftime('%Y%m%d.%H%M%S')
     config['BKG_ISOTIME'] = bkg_time_obj.strftime('%Y-%m-%dT%H:%M:%SZ')
+    config['LAND_BKG_YYYYmmddHHMMSS'] = bkg_time_obj.strftime('%Y%m%d.%H%M%S')
+    config['LAND_BKG_ISOTIME'] = bkg_time_obj.strftime('%Y-%m-%dT%H:%M:%SZ')
     if 'assim_freq' in os.environ:
         config['ATM_WINDOW_LENGTH'] = f"PT{os.environ['assim_freq']}H"
     elif 'atm_window_length' in config.keys():
         config['ATM_WINDOW_LENGTH'] = config['atm_window_length']
+    elif 'land_window_length' in config.keys():
+        config['LAND_WINDOW_LENGTH'] = config['land_window_length']
     else:
         raise KeyError("Neither $(atm_window_length) nor ${assim_freq} defined")
     # get atm window begin
-    h = re.findall('PT(\\d+)H', config['ATM_WINDOW_LENGTH'])[0]
-    win_begin = valid_time_obj - datetime.timedelta(hours=int(h)/2)
-    win_end = valid_time_obj + datetime.timedelta(hours=int(h)/2)
-    config['ATM_WINDOW_BEGIN'] = win_begin.strftime('%Y-%m-%dT%H:%M:%SZ')
-    config['ATM_WINDOW_END'] = win_end.strftime('%Y-%m-%dT%H:%M:%SZ')
-    config['BEGIN_YYYYmmddHHMMSS'] = win_begin.strftime('%Y%m%d.%H%M%S')
+    if 'atm_window_length' in config.keys():
+        h = re.findall('PT(\\d+)H', config['ATM_WINDOW_LENGTH'])[0]
+        win_begin = valid_time_obj - datetime.timedelta(hours=int(h)/2)
+        win_end = valid_time_obj + datetime.timedelta(hours=int(h)/2)
+        config['ATM_WINDOW_BEGIN'] = win_begin.strftime('%Y-%m-%dT%H:%M:%SZ')
+        config['ATM_WINDOW_END'] = win_end.strftime('%Y-%m-%dT%H:%M:%SZ')
+        config['BEGIN_YYYYmmddHHMMSS'] = win_begin.strftime('%Y%m%d.%H%M%S')
+    # get land window begin
+    elif 'land_window_length' in config.keys():
```

Comment:
3 questions: 
1. Should BEGIN_YYYYmmddHHMMSS have an ATM_ prefix, for consistency with the variables above? 
2. Is there any difference between the land and atm variables. If not, why do we need separate land variables. 
3. The land DA is applied to a restart - for DA with IAU, the restarts are only available at the start of the DA window (and not the middle). I'm not sure how the valid_time is defined - will this give us the increment at a time that we have a restart?

Reply:
1) probably good to be safe
2) I guess the question is, would there ever be a land analysis at a different resolution from atmosphere? Probably not?
3) This is something we still need to sort out, for this initial go, it will be 3DVar everywhere. When I have a solution for IAU, it will be updated appropriately.

Reply:
I think(?) that for IAU the timing of the restarts is different, so we may eventually need some options in here to change the timing of the land DA update within the window. If there's a restart available at valid_time, then we're good for now.

Reply:
@jiaruidong2017 - can you response to this?

Reply:
@ClaraDraper-NOAA I think the IAU timing of the restarts for the GTS snow DA is consistent with the atm DA, which is in the middle of the DA window. For the GTS snow DA, we use the snow depth observations from the adpsfc dump files, which are same as the atm DA to use the conventional dump or prepbufr files. Please let me know if you have different idea.

Reply:
@jiaruidong2017 Have you added the ATM prefix to BEGIN_YYYYmmddHHMMSS? See point 1 above. 

For the timing, the (possible) issue is that the land DA update is applied to a restart. So we should make the analysis time be the same as the restart time - which means it may not be the same as the atmos analysis time. For example,  for IAU the restart is at the start of the window and I think (?) the (nominal) atmos analysis time is the middle of the window. In this case the land analysis time should be the start of the window (or, the end of the previous window).

Reply:
> Have you added the ATM prefix to BEGIN_YYYYmmddHHMMSS? See point 1 above.

@ClaraDraper-NOAA I couldn't make this change, because this is belong to the atm DA. If making this change, the yaml files in the `parm/atm/` will require the corresponding changes. I don't have authority to make any changes for the atm DA. @CoryMartin-NOAA Am I correct?

Reply:
@ClaraDraper-NOAA @CoryMartin-NOAA I added an `ATM_` prefix to the `BEGIN_YYYYmmddHHMMSS` in `yamltools.py` and `parm/atm/hofx/hofx4d.yaml`. Thanks.

Reply:
> For the timing, the (possible) issue is that the land DA update is applied to a restart. So we should make the analysis time be the same as the restart time - which means it may not be the same as the atmos analysis time. For example, for IAU the restart is at the start of the window and I think (?) the (nominal) atmos analysis time is the middle of the window. In this case the land analysis time should be the start of the window (or, the end of the previous window).

@ClaraDraper-NOAA This is a key topic for adding our land analysis. We can continue this discussion before our next PR on adding the `add_increment` for land analysis into the GDASApp.